assets/js/components/consent-mode/ConsentModeSwitch.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import {
+	createInterpolateElement,
+	useState,
+	Fragment,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Switch } from 'googlesitekit-components';
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import ErrorText from '../../components/ErrorText';
+import Link from '../Link';
+import LoadingWrapper from '../LoadingWrapper';
+import ConfirmDisableConsentModeDialog from './ConfirmDisableConsentModeDialog';
+
+const { useDispatch, useSelect } = Data;
+
+export default function ConsentModeSwitch( { loading } ) {
+	const [ saveError, setSaveError ] = useState( null );
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+
+	const isConsentModeEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).isConsentModeEnabled()
+	);
+
+	const consentModeDocumentationURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getDocumentationLinkURL( 'consent-mode' )
+	);
+
+	const isSaving = useSelect( ( select ) =>
+		select( CORE_SITE ).isFetchingSaveConsentModeSettings()
+	);
+
+	const { setConsentModeEnabled, saveConsentModeSettings } =
+		useDispatch( CORE_SITE );
+
+	async function saveSettings() {
+		setSaveError( null );
+
+		const { error } = await saveConsentModeSettings();
+
+		if ( error ) {
+			setSaveError( error );
+		}
+	}
+
+	return (
+		<Fragment>
+			<div>
+				{
+					<LoadingWrapper
+						loading={ loading }
+						width="180px"
+						height="21.3px"
+					>
+						<Switch
+							label={ __(
+								'Enable consent mode',
+								'google-site-kit'
+							) }
+							checked={ isConsentModeEnabled }
+							disabled={ loading || isSaving }
+							onClick={ () => {
+								if ( isConsentModeEnabled ) {
+									setShowConfirmDialog( true );
```

Comment:
It's a bit odd that the "show confirm dialog" is confirming you want to disable Consent Mode (eg. it's a "disable dialog"). 🤷🏻‍♂️

Reply:
That is a fair point, it's a little counter intuitive. It's not tooo bad, and I see you have approved this so in the interests of moving along I think we can consider renaming it in a future iteration.